### ISSUE TITLE
MKL now free

### ISSUE THREAD

**Issue by ocaisa** [hpcugent/easybuild-easyblocks#1116]
https://software.intel.com/en-us/articles/free-mkl

Since the MKL is now free, it looks like the `silent.cfg` no longer requires an `activation_type`. An attempted build of last weeks release yields:
```
== 2017-02-23 11:11:23,156 build_log.py:147 ERROR EasyBuild crashed with an error (at ?:124 in __init__): build failed (first 300 chars): cmd "./install.sh  -s /gpfsdata/aocais/ECAM/build/imkl/2017.2.174/iimpi-system-GCC-system-2.27/l_mkl_2017.2.174/silent.cfg" exited with exitcode 2 and output:
Name "ACTIVATION_TYPE" is not valid
```

**Comment by ocaisa** [comment]
For me the following silent.cfg seemed to work for an offline install:
```
SIGNING_ENABLED=yes
PSET_INSTALL_DIR=/gpfsdata/aocais/ECAM/software/imkl/2017.2.174-iimpi-system-GCC-system-2.27
ACCEPT_EULA=accept
INSTALL_MODE=NONRPM
CONTINUE_WITH_OPTIONAL_ERROR=yes
COMPONENTS=ALL
```

**Comment by ocaisa** [comment]
Although it still tried to do `curl POST` at the end:
```
curl -X POST --max-time 5 --retry 3 -s -d @/gpfslocal/pub/tmp/install.3a3HQj/stat_data.json -o /gpfslocal/pub/tmp/install.3a3HQj/transfer.log -k https://ircws.intel.com/wsregcentertools/wsRegCenterISIP.asmx/RecordISIPData?deviceId=00000000-0000-0000-0000-000000000001
```

**Comment by boegel** [comment]
A similar issue occurs when trying to install Intel Python, see https://github.com/hpcugent/easybuild-easyconfigs/issues/3922 

I've been working on an update of the `IntelBase` easyblock to make it more permissive w.r.t. providing a license file, PR coming soon...

**Comment by ocaisa** [comment]
How about a new extravar for intelbase:
```
licensed = False
```
which could just clean out anything with activation_ in silent.cfg

**Comment by boegel** [comment]
That's exactly what I was working on, but I got stuck a bit too deep down the rabbit hole with other stuff to get Intel Python to work...

**Comment by ocaisa** [comment]
Ok, I will give it a shot with this version of MKL

**Comment by DanielTosello** [comment]
this also applies to Intel MPI, and this patch should also resolve the issue... No extra steps needed, apart from adjusting the easyconfig.

MKL is now distributed as part of XE studio also, in /compilers_and_libraries_2017/linux/mkl/.

(I just built it from the free distro though, w/ iimpi toolchain...)

**Comment by ocaisa** [comment]
I've updated the PR so you won't even need to modify the easyconfig, it'll figure out from the version that it doesn't need to check for a licence

**Comment by boegel** [comment]
this is now fixed by #1117 which updates the `imkl` and `impi` easyblocks, so let's follow up there?
